### ISSUE TITLE
Segmentation fault on Python3.6 / Linux / frozendict 2.2 on compare.

### ISSUE THREAD

**Issue by maartenbreddels** [Marco-Sulla/python-frozendict#48]
Hi,

On debian:bullseye-slim docker image, with Python 3.6, frozendict 2.2 I get a segfault, I see this when running on CI on github actions, and managed to reproduce this locally (not easy), the code that crashes is:
```
> /vaex/packages/vaex-core/vaex/dataset.py(494)__eq__()
-> return self._ids == rhs._ids
(Pdb) self._ids
frozendict.frozendict({'x': '13c1af5d9cc28fb7a08eceeaa5ef703087cc5006154be10f64188937804de3c8', 'y': '7f869c9fd6388e26a085815327a1f92cbe50cdd79e4be7c816eeaafa21b66e23'})
(Pdb) rhs._ids
frozendict.frozendict({'z': '13c1af5d9cc28fb7a08eceeaa5ef703087cc5006154be10f64188937804de3c8', 'y': '7f869c9fd6388e26a085815327a1f92cbe50cdd79e4be7c816eeaafa21b66e23'})
False
(Pdb) rhs._ids == self._ids
Fatal Python error: Segmentation fault

Current thread 0x00007f9e72bbd740 (most recent call first):
  File "<stdin>", line 1 in <module>
  File "/opt/conda/envs/vaex-dev/lib/python3.6/pdb.py", line 376 in default
  File "/opt/conda/envs/vaex-dev/lib/python3.6/cmd.py", line 216 in onecmd
  File "/opt/conda/envs/vaex-dev/lib/python3.6/pdb.py", line 418 in onecmd
  File "/opt/conda/envs/vaex-dev/lib/python3.6/cmd.py", line 138 in cmdloop
  File "/opt/conda/envs/vaex-dev/lib/python3.6/pdb.py", line 321 in _cmdloop
  File "/opt/conda/envs/vaex-dev/lib/python3.6/pdb.py", line 352 in interaction
  File "/opt/conda/envs/vaex-dev/lib/python3.6/pdb.py", line 261 in user_line
  File "/opt/conda/envs/vaex-dev/lib/python3.6/bdb.py", line 69 in dispatch_line
  File "/opt/conda/envs/vaex-dev/lib/python3.6/bdb.py", line 51 in trace_dispatch
  File "/vaex/packages/vaex-core/vaex/dataset.py", line 494 in __eq__
  File "/home/maartenbreddels/github/vaexio/vaex/tests/dataset_test.py", line 96 in test_array_rename
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/python.py", line 183 in pytest_pyfunc_call
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/python.py", line 1641 in runtest
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 162 in pytest_runtest_call
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 255 in <lambda>
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 311 in from_call
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 255 in call_runtest_hook
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 215 in call_and_report
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 126 in runtestprotocol
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/runner.py", line 109 in pytest_runtest_protocol
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/main.py", line 348 in pytest_runtestloop
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/main.py", line 323 in _main
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/main.py", line 269 in wrap_session
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/config/__init__.py", line 163 in main
  File "/opt/conda/envs/vaex-dev/lib/python3.6/site-packages/_pytest/config/__init__.py", line 185 in console_main
  File "/opt/conda/envs/vaex-dev/bin/py.test", line 11 in <module>
Segmentation fault (core dumped)
```

So it's simply a dict with strings. Downgrading to 2.1.x solves the issue for us.

I hope this is useful.

Regards,

Maarten

**Comment by Marco-Sulla** [comment]
Thank you for the report.

This should be fixed in `frozendict` 2.2.1 . Can you confirm?

The commit: https://github.com/Marco-Sulla/python-frozendict/commit/8f10b9479981a661f7d43c14153fe7463d3d9fda

**Comment by Marco-Sulla** [comment]
Long time is passed, I think I can close here.
